### ISSUE TITLE
[23.0.0] The `org.graalvm.truffle:truffle-api` JAR contains JDK-19 specific files which are compiled with JDK 20

### ISSUE THREAD

**Issue by boris-petrov** [oracle/graal#6801]
**Describe GraalVM and your environment :**
 - GraalVM version or commit id if built from source: **23.0.0**
 - CE or EE: **CE**
 - JDK version: **19**
 - OS and OS Version: **Linux**
 - Architecture: **amd64**
 - The output of `java -Xinternalversion`:
```
OpenJDK 64-Bit Server VM (19.0.2+7) for linux-amd64 JRE (19.0.2+7), built on 2023-01-18T16:17:00Z by "builduser" with gcc 12.2.1 20230111
```

**Have you verified this issue still happens when using the latest snapshot?**
No.

**Describe the issue**
Running Graal's Truffle API on JDK 19 doesn't work. I get:

```
Suppressed: org.graalvm.polyglot.PolyglotException: java.lang.UnsupportedClassVersionError: com/oracle/truffle/api/impl/JDKAccessor has been compiled by a more recent version of the Java Runtime (class file version 64.0), this version of the Java Runtime only recognizes class file versions up to 63.0
```

Which is normal because the file `META-INF/versions/19/com/oracle/truffle/api/impl/JDKAccessor.class` in the `org.graalvm.truffle:truffle-api` JAR is compiled with JDK 20.

**Code snippet or code repository that reproduces the issue**
Any code that uses Graal.

**Steps to reproduce the issue**
Any code that uses Graal.

**Expected behavior**
This file to be compiled with JDK-19 max.

**Additional context**
This makes the Truffle API unusable on JDK-19.

**Comment by chumer** [comment]
This was quite some oversight. we test these days with 20 and 17.
The fix likely be to bump the version for the JDKAccessor class to 20.

**Comment by chumer** [comment]
Well use JDK 20 or remove the folder from the jar.
this is only used to detect loom usage atm.

**Comment by boris-petrov** [comment]
I can't use JDK 20 unfortunately because Gradle still doesn't support it (fully). As for removing the folder from the JAR... that's easier said than done when using build systems like Gradle. :smile: If there are no plans for a quick-release I'll try to think of something, thanks!

**Comment by chumer** [comment]
You can also use JDK 17 or 18. Sorry for the inconvenience. 
It can take a bit until the fix lands on Maven.

**Comment by boris-petrov** [comment]
No worries. Thanks for the great work and support!

**Comment by gilles-duboscq** [comment]
I think the issue comes from this code:
https://github.com/graalvm/mx/blob/b35d493d0a802654dd0b6c2ee6d4834c14ebc38e/mx.py#L7757-L7777

I don't think we should ever use something else than `compliance` to set `-target`.

I'm not 100% sure yet how we should deal with `javaPreviewNeeded` here. Especially since if a class gets its minor java version set to 65535 for preview feature usage it can then only be used with a VM where the major version is an exact match.

/cc @dougxc

**Comment by gilles-duboscq** [comment]
My recommendation in the short term would be to ensure `javaPreviewNeeded` is not used in anything we ship. If it's possible to remove in this current truffle case, then we can just push the problem to the next time we want to use preview features in something we ship.

**Comment by chumer** [comment]
We want to detect loom and fail. If we bump to the next release e.g. 20, the detection will not work for 19.

**Comment by boris-petrov** [comment]
I believe this issue has been resolved in GraalVM 23.0.1. In it I see only version-specific files for JDK 20. Thanks all!
